### PR TITLE
create test data, unit tests and function body for `generate_return_series`

### PR DESCRIPTION
**Pull request by JohnShiuMK**

### CODE REVIEW

File: src/quanteda/generate_return_series.py [hunk 1]
```diff
@@ -52,3 +52,34 @@ def generate_return_series(
...
+        raise ValueError("annual_volatility < 0.")
+
+    if freq == 'D':
+        periods_per_year = 365
+    elif freq == 'H':
+        periods_per_year = 365 * 24
+    elif freq == 'min':
+        periods_per_year = 365 * 24 * 60
+    else:
+        raise ValueError("Invalid frequency. Use 'D' for daily, 'H' for hourly, 'min' for minutely.")
```
Comment (meretelutz) [inline src/quanteda/generate_return_series.py:66]: This is something really small/nitpicky, but minutely isn't the right word to use here. Would the suggestion below work?
```suggestion
        raise ValueError("Invalid frequency. Use 'D' for day, 'H' for hour, 'min' for minute.") 
```

File: tests/test_generate_return_series.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            expected_annual_return=0.05,
+            annual_volatility=0.2,
+            n_rows=3,
+            freq='ABC',
+            num_series=3,
+            dist='normal',
+            random_state=524,
+            start_date='2024-01-01'
+        )
+    assert str(error.value) == "Invalid frequency. Use 'D' for daily, 'H' for hourly, 'min' for minutely."
```
Comment (meretelutz) [inline tests/test_generate_return_series.py:37]: Same issue as above here. Would this work?
```suggestion
    assert str(error.value) == "Invalid frequency. Use 'D' for day, 'H' for hour, 'min' for minute"
```

**Review by meretelutz — CHANGES_REQUESTED** [review]
Looks good! Just a small issue with using 'minutely'.

File: tests/test_generate_return_series.py (hunk 2 above)
Comment (JohnShiuMK) [reply tests/test_generate_return_series.py:37]: thanks Merete

**Review by meretelutz — APPROVED** [review]
Looks great!

**Review by dorisyycai — COMMENTED** [review]
looks good to me

**Pull request merged**